src/Codegen/Builders/ObjectBuilder.hack

```diff
@@ -90,16 +90,16 @@ class ObjectBuilder extends CompositeBuilder {
     }
 
     // TODO: It should be possible to create user-defined edges which contain additional fields.
-    public static function forEdge(string $hack_type): ObjectBuilder {
-        $name = $hack_type.'Edge';
+    public static function forEdge(string $gql_type, string $hack_type, string $output_type): ObjectBuilder {
```

Comment:
We now correctly handle Hack primitives.

Reply:
I can't imagine what that could be useful for :) but I agree it's probably better to not have special cases around which exact GraphQL types are allowed where.

Reply:
I guess if you're paginating over a query that does something like `SELECT name...`. But yeah, not likely.

Reply:
That should probably be discouraged (the strength of GraphQL is that clients can query for exactly what data they need in a structured way, so returning only the name instead of a User object would violate that), but I guess it might be useful in some rare cases.